src/IAM/Domain/User/Model/Email.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Twitter\IAM\Domain\User\Model;
+
+use Twitter\IAM\Domain\User\Model\Exception\InvalidEmailException;
+
+final readonly class Email
+{
+    public function __construct(
```

Comment:
```suggestion
    private function __construct(
```
To force immutability for value object and promote contextual constructor `fromString` below.